test/e2e/specs/tools/advertising__promote.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @group calypso-pr
+ * @group jetpack-wpcom-integration
+ */
+
+import {
+	getTestAccountByFeature,
+	envToFeatureKey,
+	envVariables,
+	DataHelper,
+	SidebarComponent,
+	AdvertisingPage,
+	TestAccount,
+	MediaHelper,
+	BlazeCampaignPage,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+import { skipDescribeIf } from '../../jest-helpers';
+import { TEST_IMAGE_PATH } from '../constants';
+
+declare const browser: Browser;
+
+/**
+ * Walks through the Advertising/Blaze flow until immediately prior to the checkout.
+ *
+ * Keywords: Jetpack, Blaze, Advertising
+ */
+skipDescribeIf( envVariables.ATOMIC_VARIATION === 'private' )(
+	DataHelper.createSuiteTitle( 'Advertising: Promote' ),
+	function () {
+		const pageTitle = DataHelper.getRandomPhrase();
+		const snippet = Array( 5 ).fill( DataHelper.getRandomPhrase() ).toString();
+
+		let page: Page;
+		let testAccount: TestAccount;
+		let advertisingPage: AdvertisingPage;
+		let blazeCampaignPage: BlazeCampaignPage;
+
+		beforeAll( async () => {
+			page = await browser.newPage();
+
+			const accountName = getTestAccountByFeature( envToFeatureKey( envVariables ) );
+			testAccount = new TestAccount( accountName );
+			await testAccount.authenticate( page );
+
+			advertisingPage = new AdvertisingPage( page );
+		} );
+
+		it( 'Navigate to Tools > Advertising page', async function () {
+			if ( envVariables.ATOMIC_VARIATION === 'ecomm-plan' ) {
+				await advertisingPage.visit( testAccount.getSiteURL( { protocol: false } ) );
+			} else {
+				const sidebarComponent = new SidebarComponent( page );
+				await sidebarComponent.navigate( 'Tools', 'Advertising' );
+			}
+		} );
+
+		it( 'Click on Promote for the first post', async function () {
+			await advertisingPage.clickButton( 'Promote', { row: 0 } );
+		} );
```

Comment:
**Suggestion (non-blocking):** Since we need one post, do you think we should make on with the REST API for the rare case of right after the blog being emptied? WDYT?

Reply:
Hmm, that makes sense, let's do that.

Reply:
Added setup and teardown to ensure at least one post available.